zebra-consensus/src/block/tests.rs

```diff
@@ -4,181 +4,118 @@ use super::*;
-
-    let state_service = zebra_state::in_memory::init();
-    let mut block_verifier = super::init(state_service.clone());
-
-    // Modify the block's time
-    // Changing the block header also invalidates the header hashes, but
-    // those checks should be performed later in validation, because they
-    // are more expensive.
-    let three_hours_in_the_future = Utc::now()
-        .checked_add_signed(chrono::Duration::hours(3))
-        .ok_or("overflow when calculating 3 hours in the future")
-        .map_err(|e| eyre!(e))?;
-    block.header.time = three_hours_in_the_future;
-
-    let arc_block: Arc<Block> = block.into();
-
-    /// SPANDOC: Make sure the verifier service is ready
-    let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
-    /// SPANDOC: Try to add the block, and expect failure
-    // TODO(teor || jlusby): check error kind
-    ready_verifier_service
-        .call(arc_block.clone())
-        .await
-        .unwrap_err();
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn header_solution_test() -> Result<(), Report> {
-    header_solution().await
-}
-
-#[spandoc::spandoc]
-async fn header_solution() -> Result<(), Report> {
-    zebra_test::init();
-
-    // Service variables
-    let state_service = Box::new(zebra_state::in_memory::init());
-    let mut block_verifier = super::init(state_service.clone());
-
-    let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
-
-    // Get a valid block
-    let mut block = Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
-        .expect("block test vector should deserialize");
-
-    // This should be ok
-    ready_verifier_service
-        .call(Arc::new(block.clone()))
-        .await
-        .map_err(|e| eyre!(e))?;
+use zebra_test::transcript::{TransError, Transcript};
+
+static VALID_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
+    Lazy::new(|| {
+        let block: Arc<_> =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+                .unwrap()
+                .into();
+        let hash = Ok(block.as_ref().into());
+        vec![(block, hash)]
+    });
+
+static INVALID_TIME_BLOCK_TRANSCRIPT: Lazy<Vec<(Arc<Block>, Result<BlockHeaderHash, TransError>)>> =
+    Lazy::new(|| {
+        let mut block: Block =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+                .unwrap()
+                .into();
+
+        // Modify the block's time
+        // Changing the block header also invalidates the header hashes, but
+        // those checks should be performed later in validation, because they
+        // are more expensive.
+        let three_hours_in_the_future = Utc::now()
+            .checked_add_signed(chrono::Duration::hours(3))
+            .ok_or("overflow when calculating 3 hours in the future")
+            .map_err(|e| eyre!(e))
```

Comment:
```suggestion
            .ok_or_else(|| eyre!("overflow when calculating 3 hours in the future"))
```

Reply:
Alternatively its probably safe to use `expect` here because we never expect 3 hours from the current time to cause overflow.

Reply:
Added at https://github.com/ZcashFoundation/zebra/pull/675/commits/2ced9f7cf713b5c83637ece0874cf2630ebcc673